etna/models/tbats.py

```diff
@@ -65,7 +65,31 @@ def forecast(self, df: pd.DataFrame, prediction_interval: bool, quantiles: Itera
         return y_pred
 
     def predict(self, df: pd.DataFrame, prediction_interval: bool, quantiles: Iterable[float]) -> pd.DataFrame:
-        raise NotImplementedError("Method predict isn't currently implemented!")
+        if self._fitted_model is None or self._freq is None:
+            raise ValueError("Model is not fitted! Fit the model before calling predict method!")
+
+        y_pred = pd.DataFrame()
+        y_pred["target"] = self._fitted_model.y_hat
+        y_pred["timestamp"] = pd.date_range(end=str(self._last_train_timestamp), freq=self._freq, periods=len(y_pred))
+
+        if len(set(y_pred["timestamp"]) & set(df["timestamp"])) == 0:
```

Comment:
I guess there should be `set(df["timestamp"]) <= set(y_pred["timestamp"])`, otherwise we can't predict all the timestamps in passed dataframe

Reply:
Can't we check it in the begining and just fill "target" in passed df?